content/blog/2017/07/2017-07-06-continuousdelivery-devops-artifactory.adoc

```diff
@@ -0,0 +1,282 @@
+release versions, as part of the pipeline, by using a dedicated Maven plugin, and externalizing it into a Jenkins shared library. This can look like the following:
+
+.sl.groovy (excerpt): A Jenkins shared library, to include in Jenkins pipelines.
+[[app-listing]]
+----
+    #!/usr/bin/groovy
+    def call(args) { // <1>
+       echo "Calling shared library, with ${args}."
+       sh "mvn com.huettermann:versionfetcher:1.0.0:release versions:set -DgenerateBackupPoms=false -f ${args}"  // <2>
+    }
+----
+<1> We provide a gloabel variable/function to include it in our pipelines.
+<2> The library calls a Maven plugin, which dynamically morphs the snapshot version of a Maven project to a release version.
+
+And including it into the pipepline is then also very straight forward:
+
+.pipeline.groovy (excerpt): A stage calling a Jenkins shared library.
+[[app-listing]]
+----
+    stage('Produce RC') { // <1>
+        releaseVersion 'all/pom.xml' // <2>
+    }
+----
+<1> This stage is part of a scripted pipeline and is dedicated to morphing a Maven snapshot version into a release version, dynamically.
+<2> We call the Jenkins shared library, with a parameter pointing to the Maven POM file, which can be a parent POM.
+
+You can find the code of the underlying Maven plugin link:https://github.com/michaelhuettermann/sandbox/blob/master/versionfetcher/src/main/java/VersionFetcher/VersionFetcher.java[here].
+
+Let's now discuss how to proceed for the release candidates.
+
+=== Release Candidate (RC)
+The pipeline to promote a dev version to a RC version does contain a couple of different stages, including stages to certify the binaries (meaning labeling it or adding context information) and stages to process the concrete promotion.
+The following illustration shows the successful run of the promotion, for software version 1.0.0.
+
+image:/images/post-images/artifactory-jenkins/05.png[role=center, title="Promotion to RC. Looks like it succeeded."]
+
+We utilize Jenkins Blue Ocean that is a new user experience for Jenkins based on a personalizable, modern design that allows users to graphically create, visualize and diagnose
+delivery pipelines. Also putatively small Blue Ocean features help to boost your productivity e.g. to have log information at your fingertips and the ability to search pipelines. The stages
+to perform the promote are as follows starting with the  Jenkins pipeline stage for promoting the WAR file. Keep in mind that all scripts are parameterized, and
+the underlying call is using the JFrog CLI.
+
+.pipeline.groovy (excerpt): Staging WAR file to different logical repository
+[[app-listing]]
+----
+    stage('Promote WAR') { // <1>
+       steps { // <2>
+          sh 'jfrog rt cp --url=https://$ARTI3 --apikey=$artifactory_key --flat=true libs-release-local/com/huettermann/web/$version/ ' + // <3>
```

Comment:
Hi Daniel. thanks very much for your further feedback. I've incorporated adjustments, including optipng-optimized illustrations, and squeezed heavy-weight commits, to address all your points. I hope I did not make it even worse.